crates/shared/src/sources/balancer_v2/pools/stable.rs

```diff
@@ -55,8 +55,8 @@ impl AmplificationParameter {
     }
 
     /// This is the format used to pass into smart contracts.
-    pub fn as_u256(&self) -> U256 {
-        self.factor * self.precision
+    pub fn with_base(&self, base: U256) -> Option<U256> {
+        Some(self.factor.checked_mul(base)?.checked_mul(base)? / self.precision)
```

Comment:
This math looks quite different to what we had before. Was this causing issues?

Reply:
> Was this causing issues?

Exactly, this was the main bug I found (see PR description). The "amplification parameter" is a rational value, the "actual value" of `1.23` could be represented as `{ factor: 123, precision: 100 }` or `{ factor: 1230000, precision: 1000000 }`. Note that, with the previous implementation, they would yield **different** `as_u256` representation, despite representing the same "actual value". This was a bug since the stable math uses a [constant precision value](https://github.com/balancer/balancer-v2-monorepo/blob/227683919a7031615c0bc7f144666cdf3883d212/pkg/pool-stable/contracts/StableMath.sol#L30) of `1000`, so we want to convert the amplification parameter from  
`{ factor: 123, precision: 100 }` to `{ factor: 1230, precision: 1000 }` and yield a U256 representation of `1230`.

> This math looks quite different to what we had before

Note that if `base == precision`, then the results would be the same 😄 - except for overflow semantics.

Reply:
There was actually a booboo here - it has since been fixed.